qa-tests-backend/src/main/groovy/util/CollectorUtil.groovy

```diff
@@ -0,0 +1,109 @@
+package util
+
+import static util.Helpers.withRetry
+
+import groovy.util.logging.Slf4j
+import groovy.transform.CompileStatic
+
+import orchestratormanager.Kubernetes
+
+import com.google.protobuf.util.JsonFormat
+
+import sensor.Collector
+
+@Slf4j
+@CompileStatic
+class CollectorUtil {
+    static final String RUNTIME_CONFIG_MAP_NAME = "collector-config"
+    static final String RUNTIME_CONFIG_MAP_KEY = "runtime_config.yaml"
+
+    static final String ENABLED_VALUE = "ENABLED"
+    static final String DISABLED_VALUE = "DISABLED"
+
+    static parseJsonToProtobuf(String json) {
+        Collector.CollectorConfig.Builder builder = Collector.CollectorConfig.newBuilder()
+        JsonFormat.parser().merge(json, builder)
+        return builder.build()
+    }
+
+    static introspectionQuery(String collectorAddress, String endpoint) {
+        String uri = "http://${collectorAddress}${endpoint}"
+        URL url = new URL(uri)
+        HttpURLConnection connection = null
+
+        try {
+            connection = (HttpURLConnection) url.openConnection()
+
+            // this might be unneeded?
+            connection.setRequestMethod("GET")
+            connection.connect()
+
+            if (connection.responseCode != HttpURLConnection.HTTP_OK) {
+                throw new RuntimeException("IntrospectionQuery failed with ${connection.responseMessage}")
+            }
+            def jsonResponse = connection.inputStream.text
+            return parseJsonToProtobuf(jsonResponse)
+        } catch (Exception e) {
+            throw new RuntimeException("Error making request: ${e.message}", e)
+        } finally {
+            connection.disconnect()
```

Comment:
Do we need to call it? According to docs 
```
 * Each HttpURLConnection instance is used to make a single request
 * but the underlying network connection to the HTTP server may be
 * transparently shared by other instances. Calling the close() methods
 * on the InputStream or OutputStream of an HttpURLConnection
 * after a request may free network resources associated with this
 * instance but has no effect on any shared persistent connection.
 * Calling the disconnect() method may close the underlying socket
 * if a persistent connection is otherwise idle at that time.
 ```
 and `.text`
```
Read the content of this InputStream and return it as a String. The stream is closed before this method returns.
```

Reply:
I have removed this line.